spec/integration/support/serve_examples.rb

```diff
@@ -47,13 +47,17 @@ def test_basic_html(port)
       @machinery.run_command("pkill -f 'machinery serve' --signal 9")
     end
 
-    it "makes the system description HTML and extracted files available at the specified port" do
+    before(:each) do
       @machinery.inject_directory(
         system_description_dir,
         machinery_config[:machinery_dir],
         owner: machinery_config[:owner],
         group: machinery_config[:group]
       )
+    end
+
+    it "makes the system description HTML and extracted files available at the specified port and " \
+      "checks if the port is already in use when we run that command twice" do
```

Comment:
Wouldn't it be clearer for this that the test per se is only "checks if the port ...." ... to me it feels redundant to have the first part, plus it makes the description quite long